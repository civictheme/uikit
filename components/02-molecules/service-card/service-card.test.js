const template = 'components/02-molecules/service-card/service-card.twig';

describe('Service Card Component', () => {
  test('renders with required attributes', async () => {
    const c = await dom(template, {
      title: 'Service Card Title',
      links: [
        { text: 'Link 1', url: 'https://example.com/link1' },
        { text: 'Link 2', url: 'https://example.com/link2' },
      ],
    });

    expect(c.querySelectorAll('.ct-service-card')).toHaveLength(1);
    expect(c.querySelector('.ct-service-card__title').textContent.trim()).toEqual('Service Card Title');

    const links = c.querySelectorAll('.ct-service-card__links .ct-link');
    expect(links).toHaveLength(2);
    expect(links[0].textContent.trim()).toEqual('Link 1');
    expect(links[0].getAttribute('href')).toEqual('https://example.com/link1');
    expect(links[1].textContent.trim()).toEqual('Link 2');
    expect(links[1].getAttribute('href')).toEqual('https://example.com/link2');

    assertUniqueCssClasses(c);
  });

  test('renders with optional attributes', async () => {
    const c = await dom(template, {
      content_top: 'Top content',
      title: 'Service Card Title',
      links: [
        { text: 'Link 1', url: 'https://example.com/link1', is_new_window: true, is_external: true },
        { text: 'Link 2', url: 'https://example.com/link2' },
      ],
      content_bottom: 'Bottom content',
      theme: 'dark',
      attributes: 'data-test="true"',
      modifier_class: 'custom-class',
    });

    const element = c.querySelector('.ct-service-card');
    expect(element).not.toBeNull();
    expect(element.classList.contains('ct-theme-dark')).toBe(true);
    expect(element.classList.contains('custom-class')).toBe(true);
    expect(element.getAttribute('data-test')).toEqual('true');

    expect(c.querySelector('.ct-service-card__content-top').textContent.trim()).toEqual('Top content');
    expect(c.querySelector('.ct-service-card__content-bottom').textContent.trim()).toEqual('Bottom content');
    expect(c.querySelector('.ct-service-card__title').textContent.trim()).toEqual('Service Card Title');

    const links = c.querySelectorAll('.ct-service-card__links .ct-link');
    expect(links).toHaveLength(2);
    expect(links[0].textContent.trim()).toContain('Link 1');
    expect(links[0].getAttribute('href')).toEqual('https://example.com/link1');
    expect(links[0].getAttribute('target')).toEqual('_blank');
    expect(links[1].textContent.trim()).toEqual('Link 2');
    expect(links[1].getAttribute('href')).toEqual('https://example.com/link2');

    assertUniqueCssClasses(c);
  });

  test('does not render when title is empty', async () => {
    const c = await dom(template, {
      title: '',
      links: [
        { text: 'Link 1', url: 'https://example.com/link1' },
        { text: 'Link 2', url: 'https://example.com/link2' },
      ],
    });

    expect(c.querySelectorAll('.ct-service-card')).toHaveLength(0);
  });

  test('renders with content slots', async () => {
    const c = await dom(template, {
      content_top: 'Top content',
      title: 'Service Card Title',
      links: [
        { text: 'Link 1', url: 'https://example.com/link1' },
        { text: 'Link 2', url: 'https://example.com/link2' },
      ],
      content_bottom: 'Bottom content',
    });

    expect(c.querySelector('.ct-service-card__content-top').textContent.trim()).toEqual('Top content');
    expect(c.querySelector('.ct-service-card__content-bottom').textContent.trim()).toEqual('Bottom content');

    assertUniqueCssClasses(c);
  });
});
