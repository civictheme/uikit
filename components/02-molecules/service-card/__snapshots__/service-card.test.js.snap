// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Service Card Component does not render when title is empty 1`] = `
<div>
  


</div>
`;

exports[`Service Card Component renders with content slots 1`] = `
<div>
  

  
  <div
    class="ct-service-card ct-theme-light "
  >
    
    
    <div
      class="ct-service-card__content"
    >
      
                        
      <div
        class="ct-service-card__content-top"
      >
        Top content
      </div>
      
              
                        


  
      <h4
        class="ct-heading ct-theme-light ct-service-card__title"
      >
        Service Card Title
      </h4>
      
              
                                                                                                                                                                        

  
      <ul
        class="ct-item-list ct-item-list--vertical ct-item-list--regular  ct-service-card__links"
      >
        
                  
        <li
          class="ct-item-list__item"
        >
                          


  
  
          <a
            class="ct-link ct-theme-light"
            href="https://example.com/link1"
          >
            

    Link 1  
          </a>
          
              
        </li>
        
                        
        <li
          class="ct-item-list__item"
        >
                          


  
  
          <a
            class="ct-link ct-theme-light"
            href="https://example.com/link2"
          >
            

    Link 2  
          </a>
          
              
        </li>
        
            
      </ul>
      
              
                        
      <div
        class="ct-service-card__content-bottom"
      >
        Bottom content
      </div>
      
                  
    </div>
    
  
  </div>
  

</div>
`;

exports[`Service Card Component renders with optional attributes 1`] = `
<div>
  

  
  <div
    class="ct-service-card ct-theme-dark custom-class"
    data-test="true"
  >
    
    
    <div
      class="ct-service-card__content"
    >
      
                        
      <div
        class="ct-service-card__content-top"
      >
        Top content
      </div>
      
              
                        


  
      <h4
        class="ct-heading ct-theme-dark ct-service-card__title"
      >
        Service Card Title
      </h4>
      
              
                                                                                                                                                                        

  
      <ul
        class="ct-item-list ct-item-list--vertical ct-item-list--regular  ct-service-card__links"
      >
        
                  
        <li
          class="ct-item-list__item"
        >
                          


  
  
          <a
            aria-label="Opens in a new tab"
            class="ct-link ct-theme-dark ct-link--external"
            href="https://example.com/link1"
            target="_blank"
          >
            

    
      
      
                  
                              
          
                    
          
                                
          
                                            
          
          Link 
            <span
              class="ct-text-icon__group"
            >
              1


                                    
              <svg
                aria-hidden="true"
                class="ct-icon  "
                height="24"
                role="img"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                

                <path
                  d="M17.9199 6.62C17.8185 6.37565 17.6243 6.18147 17.3799 6.08C17.2597 6.02876 17.1306 6.00158 16.9999 6H6.99994C6.73472 6 6.48037 6.10536 6.29283 6.29289C6.1053 6.48043 5.99994 6.73478 5.99994 7C5.99994 7.26522 6.1053 7.51957 6.29283 7.70711C6.48037 7.89464 6.73472 8 6.99994 8H14.5899L6.28994 16.29C6.19621 16.383 6.12182 16.4936 6.07105 16.6154C6.02028 16.7373 5.99414 16.868 5.99414 17C5.99414 17.132 6.02028 17.2627 6.07105 17.3846C6.12182 17.5064 6.19621 17.617 6.28994 17.71C6.3829 17.8037 6.4935 17.8781 6.61536 17.9289C6.73722 17.9797 6.86793 18.0058 6.99994 18.0058C7.13195 18.0058 7.26266 17.9797 7.38452 17.9289C7.50638 17.8781 7.61698 17.8037 7.70994 17.71L15.9999 9.41V17C15.9999 17.2652 16.1053 17.5196 16.2928 17.7071C16.4804 17.8946 16.7347 18 16.9999 18C17.2652 18 17.5195 17.8946 17.707 17.7071C17.8946 17.5196 17.9999 17.2652 17.9999 17V7C17.9984 6.86932 17.9712 6.74022 17.9199 6.62Z"
                />
                

              </svg>
              

  
            </span>
            
                  
      
      
            <span
              class="ct-visually-hidden"
            >
              (Opens in a new tab/window)
            </span>
          </a>
          
              
        </li>
        
                        
        <li
          class="ct-item-list__item"
        >
                          


  
  
          <a
            class="ct-link ct-theme-dark"
            href="https://example.com/link2"
          >
            

    Link 2  
          </a>
          
              
        </li>
        
            
      </ul>
      
              
                        
      <div
        class="ct-service-card__content-bottom"
      >
        Bottom content
      </div>
      
                  
    </div>
    
  
  </div>
  

</div>
`;

exports[`Service Card Component renders with required attributes 1`] = `
<div>
  

  
  <div
    class="ct-service-card ct-theme-light "
  >
    
    
    <div
      class="ct-service-card__content"
    >
      
                    
                        


  
      <h4
        class="ct-heading ct-theme-light ct-service-card__title"
      >
        Service Card Title
      </h4>
      
              
                                                                                                                                                                        

  
      <ul
        class="ct-item-list ct-item-list--vertical ct-item-list--regular  ct-service-card__links"
      >
        
                  
        <li
          class="ct-item-list__item"
        >
                          


  
  
          <a
            class="ct-link ct-theme-light"
            href="https://example.com/link1"
          >
            

    Link 1  
          </a>
          
              
        </li>
        
                        
        <li
          class="ct-item-list__item"
        >
                          


  
  
          <a
            class="ct-link ct-theme-light"
            href="https://example.com/link2"
          >
            

    Link 2  
          </a>
          
              
        </li>
        
            
      </ul>
      
              
                        
    </div>
    
  
  </div>
  

</div>
`;
